src/main/java/org/cryptomator/cryptofs/fh/FileHeaderHolder.java

```diff
@@ -75,6 +81,9 @@ FileHeader loadExisting(FileChannel ch) throws IOException {
 			isPersisted.set(true);
 			return existingHeader;
 		} catch (IllegalArgumentException | CryptoException e) {
+			if (e instanceof AuthenticationFailedException afe) {
+				eventConsumer.accept(new DecryptionFailedEvent(path.get(), afe));
+			}
```

Comment:
As mentioned above, I'd drop this type check. The user probably doesn't care for the technical reason causing the decryption to fail.